build.gradle

```diff
@@ -88,7 +88,8 @@ dependencies {
     testImplementation 'io.github.bonigarcia:webdrivermanager:5.3.2'
     testImplementation 'org.awaitility:awaitility'
     testImplementation 'org.seleniumhq.selenium:selenium-java'
-    testImplementation 'com.h2database:h2'
+//    testImplementation 'com.h2database:h2'
+    testImplementation 'org.postgresql:postgresql'
```

Comment:
Why are we pulling H2 and replacing it with postgresql?  The beauty of H2 to is that it is an in memory database that doesn't affect the engineers db.

Reply:
We couldn't get H2 to play nice with the SQL needed for this migration. 

@bencalegari might be able to speak with more specifics on that.

Reply:
H2 will definitely make us jump through some hoops, sql statement wise, but with out H2 we will now need to make sure anyone running this test code has a db setup independently from our code - like github.

Reply:
@bencalegari which statements in Sql are having trouble being translated to H2 Sql?

Reply:
```sql
update user_files uf set new_submission_id = s.new_id from submissions s where s.id = uf.submission_id;
```
^ This fails on H2 saying `NULL not allowed for column s.id`. This is weird for many reasons, the biggest of which is that we aren't updating the submission id. Could just be a bunk error message but whether we used a placeholder for the UUID  generation function or called the H2-specific one, we got this result. H2's postgres compatibility seems insufficiently compatible. The `@Transactional` and the  `spring.jpa.hibernate.ddl-auto:create-drop` setting clean the database on test runs, leaving only the migrations. We should document the necessity to create a test db in this pr if we continue down this road. 

I really didn't want to do switch from H2 but I'm not aware of another way to phrase that necessary update in SQL. One alternative that just came to mind we didn't try was to write a [Java-based migration](https://flywaydb.org/documentation/tutorials/java#:~:text=Creating%20a%20Java%2Dbased%20migrations) but depending on how much we have to rely on underlying SQL statements instead of JDBC helpers that may not buy us enough to avoid this problem.

Reply:
Random thought:  We could also just skip that line in H2, as we have no data to actually migrate in the test scenario.